src/Middleware/RequestThrottling/src/RequestThrottlingEventSource.cs

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.Tracing;
+using System.Text;
+using System.Threading;
+using Microsoft.Extensions.Internal;
+
+namespace Microsoft.AspNetCore.RequestThrottling
+{
+    internal sealed class RequestThrottlingEventSource : EventSource
+    {
+        public static readonly RequestThrottlingEventSource Log = new RequestThrottlingEventSource();
+        private static QueueFrame _cachedNonTimerResult
```

Comment:
Property names shouldn't start with `_`

```suggestion
        private static QueueFrame CachedNonTimerResult
```

Reply:
Actually, just change this to an eagerly-initialized static field.

Reply:
Changed it back to eager-initialized, it will allocate the struct every time.

Reply:
The funny thing is you're allocating at exactly the same time you were before, but now you're allocating less.

Since QueueFrame is a struct, the lazy initialization was even more pointless than usual. Both QueueFrame and `Nullable<QueueFrame>` are structs whos values are contained directly in the parent object instead of indirectly referenced. It just that `Nullable<QueueFrame>` is even bigger since it contains both QueueFrame and the HasValue bool.

By delaying the construction of the type, it only delayed a method call (i.e. the constructor) that initialized the space already reserved for the QueueFrame. This could maybe make sense if the constructor was doing something really expensive, but then I'd argue the expensive operation should probably be moved outside of the constructor.